Purchases/Public/RCPurchases.m

```diff
@@ -599,49 +569,51 @@ - (void)updateOriginalApplicationVersionWithCompletionBlock:(RCReceivePurchaserI
                             introductoryPrice:nil
                                  currencyCode:nil
                                    completion:^(RCPurchaserInfo * _Nullable info, NSError * _Nullable error) {
-                                       [self handleUpdatedPurchaserInfo:info error:error];
+                                       [self cachePurchaserInfo:info];
                                        if (completion) {
-                                           completion(info, error);
+                                           [self dispatch:^{
+                                               CALL_AND_DISPATCH_IF_SET(completion, info, error);
+                                           }];
                                        }
                                    }];
             }];
         }
     }];
 }
 
+- (void)clearCaches {
+    [self.userDefaults removeObjectForKey:RCAppUserDefaultsKey];
+    self.cachesLastUpdated = nil;
+    self.cachedEntitlements = nil;
+}
+
 - (void)createAlias:(NSString *)alias
 {
-    [self createAlias:alias completionBlock:^(RCPurchaserInfo *info, NSError *e) {}];
+    [self createAlias:alias completionBlock:nil];
 }
 
-- (void)createAlias:(NSString *)alias completionBlock:(RCReceivePurchaserInfoBlock)completion
+- (void)createAlias:(NSString *)alias completionBlock:(RCReceivePurchaserInfoBlock _Nullable)completion
 {
     [self.backend createAliasForAppUserID:self.appUserID withNewAppUserID:alias completion:^(NSError * _Nullable error) {
         if (error == nil) {
             [self identify:alias completionBlock:completion];
         } else {
-            if (completion != nil) {
-                [self dispatch:^{
-                    completion(nil, error);
-                }];
-            }
+            CALL_AND_DISPATCH_IF_SET(completion, nil, error);
         }
     }];
 }
 
 - (void)identify:(NSString *)appUserID completionBlock:(RCReceivePurchaserInfoBlock)completion
 {
-    [self.userDefaults removeObjectForKey:RCAppUserDefaultsKey];
```

Comment:
We should still be doing this and make the changes in https://github.com/RevenueCat/purchases-ios/pull/70/files/78dd5ec6d4de1388b68dbe0c1d18b9225767df30..2d2c7b2edddffbad50dcb69b9745aed97a9e3510#r245452628